lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -110,6 +110,16 @@ def get_virtual_facts(self):
                 virtual_facts['virtualization_role'] = 'guest'
                 found_virt = True
 
+        # If docker/containerd has a custom cgroup parent, checking /proc/1/cgroup (above) might fail.
```

Comment:
Although this statement is true, I would say that this is not the only case in which the mentioned check would fail. I have been experiencing this issue for a month or so now on my up-to-date desktop Arch linux host, but only with certain Docker images, I.E. CentOS:7/8 but not Ubuntu. On an Ubuntu 20.04 host, the problem does not present itself. I have yet to pinpoint the exact change(s) that caused the breakage, but I have speculated that a modification/combination of modifications to `linux`, `docker` and/or `containerd` might have done the trick.

```
[2021-05-06T10:19:52-0700] [ALPM] upgraded containerd (1.4.4-1 -> 1.5.0-1)
[2021-05-13T08:04:26-0700] [ALPM] upgraded containerd (1.5.0-1 -> 1.5.0-2)
[2021-05-15T08:47:54-0700] [ALPM] upgraded containerd (1.5.0-2 -> 1.5.1-1)
[2021-05-23T22:12:40-0700] [ALPM] upgraded containerd (1.5.1-1 -> 1.5.2-1)

[2021-06-04T08:05:22-0700] [ALPM] upgraded docker (1:20.10.6-1 -> 1:20.10.7-1)

[2021-05-06T10:19:56-0700] [ALPM] upgraded linux (5.11.16.arch1-1 -> 5.12.1.arch1-1)
[2021-05-10T09:38:21-0700] [ALPM] upgraded linux (5.12.1.arch1-1 -> 5.12.2.arch1-1)
[2021-05-12T16:53:08-0700] [ALPM] upgraded linux (5.12.2.arch1-1 -> 5.12.3.arch1-1)
[2021-05-17T14:31:04-0700] [ALPM] upgraded linux (5.12.3.arch1-1 -> 5.12.4.arch1-2)
[2021-05-23T22:12:45-0700] [ALPM] upgraded linux (5.12.4.arch1-2 -> 5.12.6.arch1-1)
[2021-05-27T17:31:04-0700] [ALPM] upgraded linux (5.12.6.arch1-1 -> 5.12.7.arch1-1)
[2021-06-01T07:41:48-0700] [ALPM] upgraded linux (5.12.7.arch1-1 -> 5.12.8.arch1-1)
[2021-06-04T08:05:23-0700] [ALPM] upgraded linux (5.12.8.arch1-1 -> 5.12.9.arch1-1)
[2021-06-14T12:19:39-0700] [ALPM] upgraded linux (5.12.9.arch1-1 -> 5.12.10.arch1-1)
[2021-06-17T17:35:04-0700] [ALPM] upgraded linux (5.12.10.arch1-1 -> 5.12.11.arch1-1)
[2021-06-19T20:58:46-0700] [ALPM] upgraded linux (5.12.11.arch1-1 -> 5.12.12.arch1-1)
[2021-06-26T16:38:20-0700] [ALPM] upgraded linux (5.12.12.arch1-1 -> 5.12.13.arch1-2)
```